<!DOCTYPE html>
<html dir="ltr" lang="en">
<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <title>SCALCE: boosting Sequence Compression Algorithms using Locally Consistent Encoding</title>
	<meta name="keywords" content="scalce, compression, fastq, bioinformatics, reordering, lcp">
	<meta name="description" content="A tool for compressing FASTQ files">
  <link rel="stylesheet" href="combo.css">
  <link href='http://fonts.googleapis.com/css?family=Raleway:400,300,700' rel='stylesheet' type='text/css'>
  <link rel="stylesheet" href="//netdna.bootstrapcdn.com/font-awesome/4.2.0/css/font-awesome.min.css">
  
	
</head>


<body>
  <header>
    <div class="container">
      <h1>SCALCE</h1>
      <div id="menu">
      
        
        <a href="#intro">Intro</a>
      
        
        <a href="#usage">Usage</a>
      
        
        <a href="#info">Support</a>
      
        
        <a href="#releaselog">Notes</a>
      
      </div>
    </div>
  </header>


 
  <div id="main_content">     
     
      
      <section id="intro">
        <div class="container ">
          <h2 id="so-what-is-scalce">So what is SCALCE?</h2>

<p>SCALCE (<code>/skeɪlz/</code>, a.k.a. boosting <u>S</u>equence <u>C</u>ompression <u>A</u>lgorithms using <u>L</u>ocally <u>C</u>onsistent <u>E</u>ncoding) is a tool for compressing FASTQ files. It is designed specifically for the Illumina-generated FASTQ files, but supports any valid FASTQ with consistent read lengths. SCALCE was publised in <a href="http://bioinformatics.oxfordjournals.org/content/28/23/3051">Bioinformatics in October 2012</a>.</p>

<h2 id="how-do-i-get-scalce">How do I get SCALCE?</h2>

<p>Just clone our repository and issue <code>make</code> command:</p>

<pre><code>git clone https://github.com/sfu-compbio/scalce.git
cd scalce &amp;&amp; make -j
</code></pre>

<p>Alternatively, use:</p>

<pre><code>git clone git://git.code.sf.net/p/scalce/scalce \
	scalce-scalce
</code></pre>

<p>If you don’t have git, you can always fetch pre-packaged SCALCE archives:</p>

<ul>
  <li><a href="https://github.com/sfu-compbio/scalce/zipball/master">.zip package</a></li>
  <li><a href="https://github.com/sfu-compbio/scalce/tarball/master">.tar.gz package</a> </li>
</ul>

<blockquote>
  <p><strong>Note</strong>: You will need <a href="http://www.zlib.net/">zlib</a> &gt;= 1.2.6 and <a href="http://www.bzip.org/">libbzip2</a> library to compile the sources. Unfortunately, RHEL/CentOS 5.x and older come with antiquated versions of zlib, so we recommend downloading the newer version from the web. <a href="http://zlib.net/pigz/">pigz</a> is also recommended for multi-threaded mode. See <a href="#usage">Usage</a> for explanation.</p>
</blockquote>

<h2 id="how-do-i-use-scalce">How do I use SCALCE?</h2>

<p>SCALCE is invoked as following:</p>

<ul>
  <li>
    <p><strong>Compression</strong></p>

    <pre><code>scalce [input_1.fastq] -o [result]
</code></pre>

    <p>Compress <code>input_1.fastq</code> to the files <code>result.00_1.scalcen.gz</code>, <code>result.00_1.scalcer.gz</code> and <code>result.00_1.scalceq.gz</code>.</p>

    <pre><code>scalce [input_1.fastq] -r -o [result] -n [library]
</code></pre>

    <p>Compress <code>input_1.fastq</code> together with its paired end <code>input_2.fastq</code>, discarding the names and setting library name to <code>library</code>.</p>
  </li>
  <li>
    <p><strong>Decompression</strong></p>

    <pre><code>scalce [input_1.scalcen] -d -o [something.fastq]
</code></pre>

    <p>This will decompress <code>input_1.scalce\*</code> SCALCE file to <code>something.fastq</code>.</p>
  </li>
</ul>


        </div>
      </section>
    
      
      <section id="usage">
        <div class="container ">
          <h2 id="input-and-output">Input and output</h2>

<p>SCALCE is a FASTQ compression tool designed specifically for the Illumina-generated FASTQ files.
SCALCE will compress provided FASTQ files and generate three output files with extension:
- <code>.scalcen</code> for read names, 
- <code>.scalcer</code> for reads, and 
- <code>.scalceq</code> for quality scores. </p>

<p>The read length should be fixed in any of the mates in one run. This means, if you are passing 3 paired-end libraries, then the read lengths in the first mate should be fixed (i.e. 50bp). The read lengths for the second mate could be different that those of the first mate.</p>

<h2 id="requirements">Requirements</h2>

<h2 id="parameter-explanation">Parameter explanation</h2>

<h3 id="mandatory-parameters">Mandatory parameters</h3>

<ul>
  <li>
    <p><code>-o, --output</code></p>

    <p>Specifies the prefix for the output file names. Extensions and basic information will be appended. Standard output is supported in decompression mode. Use <code>-</code> to indicate standart output (i.e. <code>-o -</code>). Standard output is supported unless you use <code>-S</code> or <code>-r</code> parameters during decompression.</p>
  </li>
</ul>

<h3 id="shared-arguments-both-for-compression-and-decompression">Shared arguments (both for compression and decompression)</h3>

<ul>
  <li>
    <p><code>-r, --paired-end</code></p>

    <p>Use paired-end FASTQ files when the two ends are in seprate files. The files should be named with <code>_1</code> and <code>_2</code>. When you are passing it as input, only give <code>_1</code> file and SCALCE will replace <code>_1</code> with <code>_2</code> and read the second file.</p>

    <p>File name example: <code>XX_1.fq XX_2.fq</code></p>
  </li>
  <li>
    <p><code>-n, --skip-names [library]</code></p>

    <p>Discard original read names, and rename each read with the library prefix, such as <code>library.1</code>, <code>library.2</code> etc.  This option can improve compression rate a lot.</p>
  </li>
</ul>

<h3 id="utility-arguments">Utility arguments</h3>

<ul>
  <li>
    <p><code>-h, --help</code></p>

    <p>Prints short usage information.</p>
  </li>
  <li>
    <p><code>-v, --version</code></p>

    <p>Prints the current version of SCALCE.</p>
  </li>
</ul>

<h3 id="decompression-arguments">Decompression arguments</h3>

<ul>
  <li>
    <p><code>-d, --decompress</code></p>

    <p>Uncompress scalce files. Provide just one file name (<code>scalceq</code> for example), 
and the program will take care for the other files.</p>
  </li>
  <li>
    <p><code>-S, --split-reads [count]</code></p>

    <p>Split the output files into a bunch of files, where each file contains the
given number of the reads.</p>

    <p>Default: <strong>0</strong> (do not split)</p>
  </li>
</ul>

<h3 id="compression-arguments">Compression arguments</h3>

<ul>
  <li>
    <p><code>-B, --bucket-set-size [size][MG]</code></p>

    <p>Set bucket set size (M)egabytes or (G)igabytes. This parameter limits the main memory accessible to SCALCE. Swap files will be used to keep all neccessary data.</p>

    <p>Default: <strong>4G</strong></p>
  </li>
  <li>
    <p><code>-c, --compression [mode]</code></p>

    <p>Select compression mode. Currently available modes are:</p>

    <ul>
      <li><em>no</em> - No compression</li>
      <li><em>gz</em> - gzip compression level 6</li>
      <li><em>pigz</em> - parallelized gzip</li>
      <li><em>bz</em> - bzip2 compression</li>
    </ul>

    <p>Default: <strong>gz</strong>, or <strong>pigz</strong> if number of threads is greater than 1</p>
  </li>
  <li>
    <p><code>-A, --no-arithmetic</code></p>

    <p>Disable arithmetic coding for the quality compression and use default compression mode. This helps reduce both compression and decompression time, but the compression ratio may suffer. </p>

    <p>Default: <strong>not activated</strong></p>
  </li>
  <li>
    <p><code>-p, --lossy-percentage [percentage]</code></p>

    <p>Set lossy error percentage.</p>

    <p>Default: <strong>0</strong> (no lossy)</p>
  </li>
  <li>
    <p><code>-s, --sample-size [count]</code></p>

    <p>Specifies how many quality values should be used for statistical analysis during the lossy trasformation table creation.</p>

    <p>Default: <strong>100000</strong></p>
  </li>
  <li>
    <p><code>-t, --temp-directory [directory]</code></p>

    <p>Set directory for holding temporary files. </p>

    <p>Default: <strong>__temp__</strong></p>
  </li>
  <li>
    <p><code>-T, --threads [num]</code></p>

    <p>Specify the number of working threads</p>

    <p>Default: <em>4</em> (if the system offers less than 4 cores, number of threads will be automatically adjusted)</p>
  </li>
</ul>

<blockquote>
  <p><strong>Note</strong>: In order to take the advantage of multi-threading, <a href="http://zlib.net/pigz/">pigz</a> binary should be located within the PATH. Otherwise, you should use SCALCE with -T1 (single thread) option</p>
</blockquote>


        </div>
      </section>
    
      
      <section id="info">
        <div class="container ">
          <h2 id="contact--support">Contact &amp; Support</h2>

<p>Feel free to drop any inquiry to <a href="&#109;&#097;&#105;&#108;&#116;&#111;:&#105;&#110;&#117;&#109;&#097;&#110;&#097;&#103;&#064;">inumanag at sfu dot oh canada</a> 
or <a href="&#109;&#097;&#105;&#108;&#116;&#111;:&#102;&#104;&#097;&#099;&#104;&#064;">fhach at sfu dot oh canada</a>.</p>

<h2 id="authors">Authors</h2>

<p>DeeZ has been brought to you by:</p>

<ul>
  <li><a href="http://www.cs.sfu.ca/~fhach/personal/">Dr. Faraz Hach</a></li>
  <li><a href="http://www.sfu.ca/~inumanag">Ibrahim Numanagić</a></li>
  <li><a href="http://www.cs.bilkent.edu.tr/~calkan/">Dr. Can Alkan</a></li>
  <li><a href="http://www.cs.sfu.ca/~cenk/">Dr. S. Cenk Şahinalp</a></li>
</ul>

<p>from the <a href="http://compbio.cs.sfu.ca">Lab for Computational Biology</a> at <a href="http://www.sfu.ca">Simon Fraser University</a>, <a href="http://eichlerlab.gs.washington.edu/">Eicher Lab</a> at <a href="http://www.washington.edu">University of Washington</a>, and <a href="http://www.cs.bilkent.edu.tr/~calkan/compgen/">Alkan Lab</a> at <a href="http://www.bilkent.edu.tr/">Bilkent University</a>.</p>

<h2 id="licence">Licence</h2>

<p>Copyright (c) 2011 - 2012, Simon Fraser University. All rights reserved.</p>

<p>Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:</p>

<ul>
  <li>Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.</li>
  <li>Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.</li>
  <li>Neither the name of the Simon Fraser University nor the names of its contributors may be used to endorse or promote products derived from this software without specific prior written permission.</li>
</ul>

<p>THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT	LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR	A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR	CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,	EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,	PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR	PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF	LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING	NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS	SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.</p>

        </div>
      </section>
    
      
      <section id="releaselog">
        <div class="container ">
          <h2 id="release-notes">Release notes</h2>

<ul>
  <li><em>(20-May-2013)</em> <strong>SCALCE version 2.7 release</strong>
    <ul>
      <li>Bugfixes (no-arithmetic fix)</li>
    </ul>
  </li>
  <li><em>(13-May-2013)</em> <strong>SCALCE version 2.6 release</strong>
    <ul>
      <li>Bugfixes</li>
    </ul>
  </li>
  <li><em>(02-Apr-2013)</em> <strong>SCALCE version 2.5 release</strong>
    <ul>
      <li>Read splitting supported</li>
      <li>Standard output during decompression supported</li>
      <li>Bugfixes</li>
    </ul>
  </li>
  <li><em>(25-Mar-2013)</em> <strong>SCALCE version 2.4 release</strong>
    <ul>
      <li>Auto-pigz detection</li>
      <li>Bugfixes</li>
    </ul>
  </li>
  <li><em>(10-Sep-2012)</em> <strong>SCALCE version 2.3 release</strong>
    <ul>
      <li>Decompression speed improvements</li>
    </ul>
  </li>
  <li><em>(25-Jul-2012)</em> <strong>SCALCE version 2.2 release</strong>
    <ul>
      <li>Speed improvements</li>
      <li>Arithmetic coding for qualities is now optional</li>
      <li>Multiple bug fixes</li>
    </ul>
  </li>
  <li><em>(06-Jun-2012)</em> <strong>SCALCE version 2.1 release</strong>
    <ul>
      <li>Better compression of reads</li>
      <li>Arithmetic coding for qualities</li>
      <li>Multiple bug fixes</li>
    </ul>
  </li>
  <li><em>(02-Mar-2012)</em> <strong>SCALCE version 1.4 release</strong>
    <ul>
      <li>Serious data loss when using multithreading bug fixed</li>
    </ul>
  </li>
  <li><em>(20-Feb-2012)</em> <strong>SCALCE version 1.3 release</strong>
    <ul>
      <li>Various bug fixes</li>
    </ul>
  </li>
  <li><em>(17-Feb-2012)</em> <strong>SCALCE version 1.2 release</strong>
    <ul>
      <li>Various bug fixes</li>
    </ul>
  </li>
  <li><em>(08-Feb-2012)</em> <strong>SCALCE version 1.1 release</strong>
    <ul>
      <li>OpenMP support</li>
      <li>pigz support</li>
    </ul>
  </li>
  <li><em>(06-Dec-2011)</em> <strong>SCALCE version 1.0 release</strong>
    <ul>
      <li>Initial release</li>
    </ul>
  </li>
</ul>

        </div>
      </section>
    
  </div>
    
      <div id="footer" class="section text-white">
      <div class="container">
        
        <p><sub>© 2011–2015 <a href="http://compbio.cs.sfu.ca">Lab for Computational Biology at SFU</a>. All rights reserved. Based upon Github’s Hack theme and <a href="http://t413.com/SinglePaged/">SinglePaged</a> <a href="http://jekyllrb.com">Jekyll</a> theme.</sub></p>


      </div>
    </div>

</body>
</html>

