---
title: "Introduction"
---

# Introduction

### So what is SCALCE?

SCALCE (`/skeÉªlz/`, a.k.a. boosting <u>S</u>equence <u>C</u>ompression <u>A</u>lgorithms using <u>L</u>ocally <u>C</u>onsistent <u>E</u>ncoding) is a tool for compressing FASTQ files. It is designed specifically for the Illumina-generated FASTQ files, but supports any valid FASTQ with consistent read lengths. SCALCE was publised in [Bioinformatics in October 2012][3].


### How do I get SCALCE?

Just clone our repository and issue `make` command:

	git clone https://github.com/sfu-compbio/scalce.git
	cd scalce
	make download
	make

If you have issues with compiling, please try our CentOS 7-compiled x86_64 binary.

	git clone https://github.com/sfu-compbio/scalce.git binary

If you don't have git, you can always fetch pre-packaged SCALCE archives:

-	[.zip package](https://github.com/sfu-compbio/scalce/zipball/master)
-	[.tar.gz package](https://github.com/sfu-compbio/scalce/tarball/master) 
-	[Binary package](https://github.com/sfu-compbio/scalce/tarball/binary) 

> **Note**: You will need [zlib](http://www.zlib.net/) >= 1.2.6 and [libbzip2](http://www.bzip.org/) library to compile the sources. Unfortunately, RHEL/CentOS 5.x and older come with antiquated versions of zlib, so we recommend downloading the newer version via `make download`. [pigz](http://zlib.net/pigz/) is also recommended for multi-threaded mode. See [Usage](#usage) for explanation.

> **Note**: SCALCE prior to version v2.7 does not support variable read lengths. Starting with v2.8, *experimental support* for varable read lengths has been added. In order to use it, please compile with `make -j pacbio` and use `scalce-pacbio` binary to run SCALCE. All options for SCALCE are as well valid for scalce-pacbio.
Please note that SCALCE is not designed for very long reads (e.g. PacBio, Nanopore), and thus the compression performance might not be ideal.

### How do I use SCALCE?

SCALCE is invoked as following:

#### Compression
		
	scalce [input_1.fastq] -o [result]
		
will compress `input_1.fastq` to the files `result.00_1.scalcen.gz`, `result.00_1.scalcer.gz` and `result.00_1.scalceq.gz`.  

<br/>

	scalce [input_1.fastq] -r -o [result] -n [library]

will compress `input_1.fastq` together with its paired end `input_2.fastq`, discarding the names and setting library name to `library`.

#### Decompression
		
	scalce [input_1.scalcen] -d -o [something.fastq]
	
will decompress `input_1.scalce\*` SCALCE file to `something.fastq`.

[3]: http://bioinformatics.oxfordjournals.org/content/28/23/3051
